### ISSUE TITLE
/commit not rebuilding repo map

### ISSUE THREAD

**Issue by ErykCh** [paul-gauthier/aider#729]
### Issue

Hi,

I added files to the project, requested in #728 and I did /commit.

But it seems that /commit is not updating repo map because I'm not able to add new files using /add

/ls also not display them

**Comment by ErykCh** [comment]
I use Visual Studio to preview code. And I did one refactoring outside Aider. And it looks like VS didn't add these files to the metadata kept in the .git directory

Because it looks like Aider uses data saved in .git and not actual files in the system.

It helped to do:
git update-index --really-refresh

so it looks like doing /commit in Aider after doing refactoring in an external tool doesn't update all files

**Comment by paul-gauthier** [comment]
Thanks for trying aider and filing this issue.

Aider can only work with files that are part of your git repo. If you create new files, you need to `git add` them before aider can work with them.

**Comment by ErykCh** [comment]
I think it would be helpful if Aider verified the state of the repository before sending a query to the model.

It should say that in the meantime someone has modified the files and its status is not up to date.

It's best if /commit not only commits, but also adds all new files to the project. Then you can change it to some /sync command and then the description that if you make modifications outside the aider, you must synchronize the Aider before you start asking further questions.

**Comment by paul-gauthier** [comment]
Aider always uses the current state of the repo.

**Comment by ErykCh** [comment]
Yes, but the question is if Aider can check if current state of the repo is synchronized with file system?

**Comment by paul-gauthier** [comment]
I'm not sure I understand your question.

**Comment by ErykCh** [comment]
yeah, bots still need to evolve a bit as they respond to issues added by users on github.

Can you tell me your bot's name?

**Comment by paul-gauthier** [comment]
This tool is called aider.

**Comment by ErykCh** [comment]
Nie.
Well, I will say that he handles tickets quite well, he rarely gets lost. Unless you or the bot respond alternately :-)

**Comment by paul-gauthier** [comment]
I'm going to close this issue for now, but feel free to add a comment here and I will re-open or file a new issue any time.
